### DISCUSSION TITLE
what is the relationship between dpgen and dpgen2

### DISCUSSION THREAD

**Discussion by scott-5** [deepmodeling/dpgen#1618] [Q&A] [answered]
Now dpgen2 can basically replace dpgen. 
I am very curious that
1. is dpgen2 a upgrade version of dpgen? And dpgen will be deprecated in the future?
or
2. dpgen2 is a parallel relationship of dpgen that applies to different user groups?

**Comment by njzjz** [comment, marked as answer]
dpgen2 is not based on dpgen. Not all dpgen features have been implemented in dpgen2.

**Reply by scott-5** [reply]
I don't know if my understanding is correct. The development orientation of dpgen2 is to implement the basic functions of dpgen in a cloud-native environment. But in fact, dpgen2 in debug mode can also be used in a cluster environment. For some beginners, there may be a question: should I choose dpgen or dpgen2?
@zjgemi Could you give some suggestions?

**Reply by njzjz** [reply]
The aim of dpgen2 is certainly to replace dpgen. However, both dpgen and dpgen2 lack enough developers...

**Reply by scott-5** [reply]
Got it. Thanks to the developers for their efforts. In fact, dpgen is relatively mature and excellent.
